West-Midlands-Community | Matthew Law | Module-User-Focused-Data | WEEK 2

matthewlawpixel opened the pull request:
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

This is my initial commit.

## Questions

I have created a form but i noticed that all elements are stuck together, how can i separate them?

rawrnoodles committed:
initial commit

rawrnoodles committed:
CSS Styling Change, Min Max Date Change, Background Change

cjyuan commented:
1. The shirt image is currently a foreground and not a background image. You should use CSS to make it a background image so that it won't interfere with the content in the foreground. (Currently, you are forcing the form to appear on top of an image element, which is a weird way to accomplish the effect).

2. Avoid using `position: absolute`; if an element with such property has siblings, it will disrupt the flow of the elements in the page. (If you switch the image to background, you should not need to use `position: absolute`).

3. The `for="..."` attribute in all `label` elements (except for date) are used incorrectly. 

4. Can you try making the text fields the same width as the select elements?

5. You can try using the `pattern` attribute to specify what value is accepted as a name. Right now two spaces will also be accepted as valid name. 

6. Make changes based on Lighthouse's suggestions so that your page can score 100 in Accessibilities.

matthewlawpixel commented:
<img width="1392" alt="Screenshot 2024-10-25 at 21 43 11" src="https://github.com/user-attachments/assets/52abc3a0-215c-4174-ae35-0ac54f8c6649">

matthewlawpixel commented:
Thank you CJ for the feedback. I have removed both the empty div tag and divider class from my project

rawrnoodles committed:
1)Changed image to background image

2)Removed position absolute as requested.

3) fixed for "..." attribute by adding id

4) Changed all form fields to be the same width

5) Added pattern attribute to form.

6) Achieved 100% for accessibility on lighthouse page analysis.

matthewlawpixel commented:
Thank you so much @cjyuan for taking the time to review my project. I have made the changes based on your suggestions as follows:

1)Changed image to background image.

2)Removed position absolute as requested.

3) fixed for "..." attribute by adding id attribute.

4) Changed all form fields to be the same width.

5) Added pattern attribute to form.

6) Achieved 100% for accessibility on lighthouse page analysis.

cjyuan commented:
Page looks great!, and all acceptance criteria satisfied.

The background image took a while to load. Any reason to pick an image that high in resolution? Also, you may want to try converting image to WEBP format. It is better than JPEG in terms of image compression ratio.

rawrnoodles committed:
Changed to webp image and resolution of the image is scaled down.

matthewlawpixel commented:
Yes sorry i forgot to change convert and compress the image. thank you for reminding me @cjyuan. I've changed the image to one of a lower resolution, the page should load much faster now. :)
